Remove default Lifetime::Timeframe when instanciating an Output

ktf opened the pull request:
Remove default Lifetime::Timeframe when instanciating an Output

This lifetime field will be removed soon, and this is unambiguous, because
it's in any case the default.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/AliceO2Group/AliceO2/pull/12399).
* #12401
* #12400
* __->__ #12399

alibuild commented:
Error while checking build/O2/fullCI for 917237204dbb86b0835f867c9150f17054dc5178 at 2023-12-05 13:27:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
/sw/SOURCES/O2/12399-slc8_x86-64/0/Utilities/DataSampling/src/Dispatcher.cxx:117:37: error: no matching function for call to 'o2::framework::Output::Output(<brace-enclosed initializer list>)'
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12399/917237204dbb86b0835f867c9150f17054dc5178/build_O2_fullCI/pretty.html).

ktf committed:
Remove default Lifetime::Timeframe when instanciating an Output

This lifetime field will be removed soon, and this is unambiguous, because
it's in any case the default.
